crosschain-resolver/contracts/IMetadataResolver.sol

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+interface IMetadataResolver {
+    /**
+     * @notice Get metadata about the CCIP Resolver
+     * @dev This function provides metadata about the CCIP Resolver, including its name, coin type, GraphQL URL, storage type, and encoded information.
+     * @param name The domain name in format (dnsEncoded)
+     * @return name The name of the resolver ("CCIP RESOLVER")
+     * @return coinType Resolvers coin type (60 for Ethereum)
+     * @return graphqlUrl The GraphQL URL used by the resolver
+     * @return storageType Storage Type (0 for EVM)
+     * @return storageLocation The storage identifier. For EVM chains, this is the address of the resolver contract.
+     * @return context can be l2 resolver contract address for evm chain but can be any l2 storage identifier for non evm chain
```

Comment:
This also needs clearer documentation. What do I do with this?

Reply:
The `context` used to point to the owner address for dm3's version of resolver. In the delegatable resolver version, it's  redundant with `storageLocation`, but I kept it [because our l2 subgraph uses context for the schema](https://github.com/makoto/ens-l2-delegatable-resolver-subgraph/blob/master/schema.graphql#L2) . We could get rid of `context` if we decide that l2 `domain` id always consists of `resolveraddress-namehash`.

Reply:
Right, but the docs need to say this. :)